app/controllers/appointments_controller.rb

```diff
@@ -50,10 +50,21 @@ def new
     @appointment = Appointment.copy_or_new_by(params[:copy_from])
   end
 
+  def preview
+    @appointment = Appointment.new(create_params.merge(agent: current_user))
+    @appointment.assign_to_guider
+    if @appointment.valid?
+      render :preview
+    else
+      render :new
+    end
+  end
+
   def create
     @appointment = Appointment.new(create_params.merge(agent: current_user))
     @appointment.assign_to_guider
-    if @appointment.save
+
+    if !edit_appointment? && @appointment.save
```

Comment:
Can we remove the negation? If we switch the helper to something like this instead:

```ruby
def creating?
  params[:edit_appointment]
end
```